### PR TITLE
[presto-cli] Return a proper return code on exit

### PR DESCRIPTION
**Pull request by Saurabh2993**
Merged Nezih's code with latest presto code. Created test cases to check 'stop-on-error' argument which stops execution of a file if a query fails, and returns an exit code.

### CODE REVIEW

**Comment by Saurabh2993** [comment]
Hi Guys,
Any updates on the branch?

**Comment by nezihyigitbasi** [comment]
we didn't have a chance to look at it unfortunately, but it's in my backlog.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java [hunk 1]
```diff
@@ -141,6 +141,30 @@ public void shouldRunQueryFromFile()
...
 
+    @Test(groups = CLI, timeOut = TIMEOUT)
+    public void shouldStopOnError()
+            throws IOException, InterruptedException
+    {
+        File temporayFile = File.createTempFile("test-sql", null);
+        temporayFile.deleteOnExit();
+        Files.write("select * from hive.default.nations;\nselect * from hive.default.nation;\n", temporayFile, UTF_8);
+
+        launchPrestoCliWithServerArgument("--file", temporayFile.getAbsolutePath());
```
[the remark this answers is not in the document]
Comment (nezihyigitbasi) [reply presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java]: Please also assert on the exit value of the process against the status codes we return above.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java [hunk 2]
```diff
@@ -141,6 +141,30 @@ public void shouldRunQueryFromFile()
...
+        assertThat(trimLines(presto.readRemainingOutputLines())).isEmpty();
+    }
+
```
Comment (nezihyigitbasi) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java]: Please add tests for asserting on presto cli exit code for success/failure cases.

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestingPrestoClient.java [hunk 3]
```diff
@@ -44,7 +44,8 @@
 import static com.google.common.collect.Iterables.transform;
 import static java.util.Objects.requireNonNull;
 
-public abstract class AbstractTestingPrestoClient<T>
+public abstract class
+        AbstractTestingPrestoClient<T>
```
Comment (nezihyigitbasi) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestingPrestoClient.java]: unnecessary change.

**Comment by electrum** [comment]
I made a number of changes and put up a new PR to test Travis. Will have Nezih take a look tomorrow and merge after that. Thanks for your work on getting this ready.

**Comment by findepi** [comment]
Superseded by #9227
